src/web/components/Core/PagingTool.tsx

```diff
@@ -1,70 +1,46 @@
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { useEffect, useState } from 'react';
 
+import {
+  getPageNumberOptions,
+  getRowsPerPageOptions,
+  getShowingRowsText,
+  RowsPerPageValues,
+} from './PagingToolHelper';
 import { SelectDropdown, SelectOption } from './SelectDropdown';
 
 import './PagingTool.scss';
 
-const rowsPerPageValues = [10, 25, 50, 100, 250] as const;
-
-export type RowsPerPageValues = (typeof rowsPerPageValues)[number];
-
 type PagingProps = Readonly<{
   numberTotalRows: number;
   onChangeRows: (currentPageNumber: number, currentRowsPerPage: RowsPerPageValues) => void;
-  rowsPerPageTitle?: string;
   initialRowsPerPage?: RowsPerPageValues;
   initialPageNumber?: number;
 }>;
 
 export function PagingTool({
   numberTotalRows,
   onChangeRows,
-  rowsPerPageTitle = 'Rows Per Page',
   initialRowsPerPage = 10,
   initialPageNumber = 1,
 }: PagingProps) {
-  const initialPageNumberOptions = Array.from(
-    { length: Math.ceil(numberTotalRows / initialRowsPerPage) },
-    (_, index) => index + 1
-  ).map((number) => ({
-    name: number.toString(),
-    id: number,
-  }));
+  const initialPageNumberOptions = getPageNumberOptions(numberTotalRows, initialRowsPerPage);
+  const rowsPerPageOptions = getRowsPerPageOptions(numberTotalRows);
+  const initialRowsPerPageOption = rowsPerPageOptions.find(
+    (number) => number.id === initialRowsPerPage
+  );
 
   const [rowsPerPage, setRowsPerPage] = useState<RowsPerPageValues>(initialRowsPerPage);
   const [pageNumber, setPageNumber] = useState<number>(initialPageNumber);
   const [pageNumberOptions, setPageNumberOptions] =
     useState<SelectOption<number>[]>(initialPageNumberOptions);
 
   useEffect(() => {
-    setPageNumberOptions(
-      Array.from(
-        { length: Math.ceil(numberTotalRows / initialRowsPerPage) },
-        (_, index) => index + 1
-      ).map((number) => ({
-        name: number.toString(),
-        id: number,
-      }))
-    );
+    setPageNumberOptions(getPageNumberOptions(numberTotalRows, initialRowsPerPage));
     setPageNumber(initialPageNumber);
     setRowsPerPage(initialRowsPerPage);
   }, [numberTotalRows, initialRowsPerPage, initialPageNumber]);
 
-  const rowsPerPageOptions =
-    numberTotalRows > rowsPerPageValues[0]
-      ? rowsPerPageValues
-          .filter((value) => value <= numberTotalRows)
-          .map((value) => ({
-            name: value.toString(),
-            id: value,
-          }))
-      : [{ name: rowsPerPageValues[0].toString(), id: rowsPerPageValues[0] }];
-
-  const initialRowsPerPageOption = rowsPerPageOptions.find(
-    (number) => number.id === initialRowsPerPage
-  );
-
   const onIncreasePageNumber = () => {
     if (pageNumber < Math.ceil(numberTotalRows / rowsPerPage)) {
```

Comment:
nit: could we still extract this condition please? something like:
```
export const canIncreasePageNumber = (pageNumber, numberTotalRows, rowsPerPage) => {
  return pageNumber < Math.ceil(numberTotalRows / rowsPerPage);
};
```

Reply:
Done